### ISSUE TITLE
Add support 'sklearn.neighbors._nca.NeighborhoodComponentsAnalysis'

### ISSUE THREAD

**Issue by jaehyeongAN** [microsoft/hummingbird#585]
[sklearn.neighbors._nca.NeighborhoodComponentsAnalysis](https://scikit-learn.org/stable/modules/generated/sklearn.neighbors.NeighborhoodComponentsAnalysis.html) is an algorithm that uses a technique similar to k nearest neighbors to find a space in which neighborhoods of points sharing the same labels are tighter than points with different labels. 
NCA  is very powerful if using with K-Nearest Neighbors. So. recently, NCA+KNN pipeline is used in many ml projects.

please, support NeighborhoodComponentsAnalysis model.

**Comment by interesaaat** [comment]
Hi @jaehyeongAN! I can take a look at this and let you know.
